src/main/bash/sdkman-init.sh

```diff
@@ -82,7 +82,7 @@ for f in "${scripts[@]}"; do
 	source "$f"
 done
 IFS="$OLD_IFS"
-unset scripts f
+unset OLD_IFS scripts f
```

Comment:
Why unset `OLD_IFS` at all if it is no longer being used?

Reply:
`OLD_IFS` is still used. It used to be used in 2 places but is now just used in 1.

Reply:
Is it worth cutting over to the new way in the other place too for consistency? (Only if this isn't a huge thing to do, of course). Then we don't have to worry about managing this nasty global state in the shell.

Reply:
I would have if this syntax would work where it is still used, but it won't.

The place where `IFS=$'\n'` is needed is:

```bash
scripts=($(find "${SDKMAN_DIR}/src" "${SDKMAN_DIR}/ext" -type f -name 'sdkman-*'))
```

You can't use `<variable>=<value> ` in the same command a permanent assignment.

The shells just don't allow it.

Maybe it could work inside a command substitution.

I'm actually trying to get a glob replacement for the `find` command to work.  The glob wouldn't use `IFS` at all, and wouldn't use an external command like `find`.  It seems to work from the command line, but not in the Gradle tests.  I imagine it has something to do with the bash environment settings, but I don't know what yet.

I'll add a PR for that once I get it working (and I'll change as many `find` usages to globs as I can easily do).

Reply:
Looks like it works inside the command substitution.  Don't accept this PR yet.  Will push an update in a few minutes.